MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,517 @@
+﻿using System;
+using System.Collections.Generic;
+using MinecraftClient.Mapping;
+using MinecraftClient.Inventory;
+using System.Threading;
+
+namespace MinecraftClient.ChatBots
+{
+    /// <summary>
+    /// A row of blocks that will be mined
+    /// </summary>
+    public class Row
+    {
+        private List<Location> blocksInRow;
+
+        public List<Location> BlocksToMine
+        {
+            get { return blocksInRow; }
+        }
+
+        /// <summary>
+        /// Initialize a row of blocks
+        /// </summary>
+        /// <param name="bIL"> Enter a list of blocks </param>
+        public Row(List<Location> bIL = null)
+        {
+            blocksInRow = bIL ?? new List<Location>();
+        }
+    }
+
+    /// <summary>
+    /// Several rows are summarized in a layer
+    /// </summary>
+    public class Layer
+    {
+        private List<Row> rowsToMine;
+
+        public List<Row> RowsToMine
+        {
+            get { return rowsToMine; }
+        }
+
+        /// <summary>
+        /// Add a new row to this layer
+        /// </summary>
+        /// <param name="givenRow"> enter a row that should be added </param>
+        /// <returns> Index of the last row </returns>
+        public int addRow(Row givenRow = null)
```

Comment:
Please use `PascalCase` naming style for C# method.
```suggestion
        public int AddRow(Row givenRow = null)
```